javascript/ql/lib/semmle/javascript/ApiGraphs.qll

```diff
@@ -931,6 +950,21 @@ module API {
       )
     }
 
+    pragma[nomagic]
+    private int firstSpreadIndex(InvokeExpr expr) {
+      result = min(int i | expr.getArgument(i) instanceof SpreadElement)
+    }
+
+    private predicate spreadArgumentPassing(TApiNode base, int i, DataFlow::Node spreadArray) {
+      exists(DataFlow::Node use, DataFlow::SourceNode pred, int bound, InvokeExpr invoke |
+        use(base, use) and
+        pred = trackUseNode(use, _, bound, "") and
+        invoke = pred.getAnInvocation().asExpr() and
+        i = firstSpreadIndex(invoke) and
+        spreadArray = invoke.getArgument(i - bound).(SpreadElement).getOperand().flow()
```

Comment:
I having a hard time figuring out whether the arithmetic here is correct.  

Could you add a test with some bound arguments?

Reply:
Good catch, there was indeed a bug here. Added a test and made the code easier to read.